### PR TITLE
[TRA-111] add vault query

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
add vault query

### Test Plan
unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: proto/dydxprotocol/vault/query.proto [hunk 1]
```diff
@@ -22,3 +28,25 @@ message QueryParamsRequest {}
...
+}
+
+// QueryVaultResponse is a response type for the Vault RPC method.
+message QueryVaultResponse {
+  VaultId vault_id = 1 [ (gogoproto.nullable) = false ];;
+  dydxprotocol.subaccounts.SubaccountId subaccount_id = 2 [ (gogoproto.nullable) = false ];;
+  uint64 equity = 3;
+  uint64 inventory = 4;
+  uint64 total_shares = 5;
+  repeated OwnerShares all_owner_shares = 6 [ (gogoproto.nullable) = false ];;
```
Comment (vincentwschau) [inline proto/dydxprotocol/vault/query.proto:45]: nit: Might be better to have a separate paginated query for this, considering there could be a lot of owners.

File: proto/dydxprotocol/vault/query.proto [hunk 2]
```diff
@@ -22,3 +28,26 @@ message QueryParamsRequest {}
...
+
+// QueryVaultResponse is a response type for the Vault RPC method.
+message QueryVaultResponse {
+  VaultId vault_id = 1 [ (gogoproto.nullable) = false ];
+  dydxprotocol.subaccounts.SubaccountId subaccount_id = 2
+      [ (gogoproto.nullable) = false ];
+  uint64 equity = 3;
+  uint64 inventory = 4;
+  uint64 total_shares = 5;
+  repeated OwnerShares all_owner_shares = 6 [ (gogoproto.nullable) = false ];
```
Comment (vincentwschau) [inline proto/dydxprotocol/vault/query.proto:46]: nit: Might be better to have a separate paginated query for this considering there may be many owners.

File: proto/dydxprotocol/vault/query.proto (hunk 1 above)
Comment (tqin7) [reply proto/dydxprotocol/vault/query.proto]: makes sense. created a ticket to add a separate query for owner shares. let me remove owner shares for now

File: proto/dydxprotocol/vault/query.proto (hunk 2 above)
Comment (tqin7) [reply proto/dydxprotocol/vault/query.proto:46]: makes sense. created a ticket to add a separate query for owner shares. let me remove owner shares for now

Comment (vincentwschau) [reply proto/dydxprotocol/vault/query.proto:46]: Non-blocking, I see that the owner shares are stored as only integers, and not a struct with both the owner + the shares so it won't be as simple as using `query.Paginate` on the store.

Comment (tqin7) [reply proto/dydxprotocol/vault/query.proto]: Yep will add a separate query for owner shares in phase 2

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**